src/IO/H5/Python/ExtendConnectivityData.py

```diff
@@ -0,0 +1,56 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+
+import click
+import rich
+
+import spectre.IO.H5 as spectre_h5
+
+
+@click.command()
+@click.argument(
+    "filename",
+    required=True,
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+    # help="name of the file",
+)
+@click.option(
+    "--subfile-name",
+    "-d",
+    required=True,
+    type=str,
+    help="subfile name of the volume file in the H5 file (omit file extension)",
+)
+def extend_connectivity_data_command(filename, subfile_name):
+    """Extend the connectivity inside a single HDF5 volume file for some
+    SpECTRE evolution, in order to fill in gaps between elements. Intended
+    to be used as a post-processing routine to improve the quality of
+    visualizations. Note: This does not work with subcell or AMR systems, and
+    the connectivity only extends *within* each block and not between them
+    """
+    # Ensure that the format of the subfile is correct
+    if subfile_name.endswith(".vol"):
+        subfile_name = subfile_name[: len(subfile_name) - 4]
+    if subfile_name[0] != "/":
+        subfile_name = "/" + subfile_name
+    # Read the h5 file and extract the volume file from it
+    h5_file = spectre_h5.H5File(filename, "r+")
+    vol_file = h5_file.get_vol(subfile_name)
+
+    observation_ids = vol_file.list_observation_ids()
+    dim = len(vol_file.get_extents(observation_ids[0])[0])
```

Comment:
I was looking at spectre documentation, and this is what it says for get_dimension:
Read the dimensionality of the grids. Note : This is the dimension of the grids as manifolds, not the dimension of the embedding space. For example, the volume data of a sphere is 2-dimensional, even though each point has an x, y, and z coordinate

Reply:
The `get_extents` function returns the number of grid points in each dimension of the grid, so that's exactly the "dimension of the grid as manifold".